src/globalsearch/spotifysearchprovider.cpp

```diff
@@ -90,30 +89,60 @@ void SpotifySearchProvider::SearchFinishedSlot(
   PendingState state = it.value();
   queries_.erase(it);
 
+  QMap<std::string, std::string> album_dedup;
   ResultList ret;
-  for (int i = 0; i < response.result_size(); ++i) {
-    const pb::spotify::Track& track = response.result(i);
-
-    Result result(this);
-    SpotifyService::SongFromProtobuf(track, &result.metadata_);
-
-    ret << result;
-  }
 
   for (int i = 0; i < response.album_size(); ++i) {
     const pb::spotify::Album& album = response.album(i);
 
+    QHash<QString, int> artist_count;
+    QString artist, majority_artist;
+    int majority_count = 0;
+
+    // Choose an album artist
+    for (int j = 0; j < album.track_size(); ++j) {
+      for (int k = 0; k < album.track(j).artist_size(); ++k) {
+        artist = QStringFromStdString(album.track(j).artist(k));
+        if (artist_count.contains(artist)) {
+          artist_count[artist]++;
+        } else {
+          artist_count[artist] = 1;
```

Comment:
I don't think you need the else if/else, because IIRC trying to access an element that doesn't exist will create it with default value (0). That will make the code smaller.
That being said, maybe this behavior is a bit confusing and it's better to keep a the code as it is.